tests/unit/Types/ContextFactoryTest.php

```diff
@@ -88,6 +88,26 @@ public function testReadsAliasesFromProvidedNamespaceAndContent()
 
             $this->assertSame($expected, $context->getNamespaceAliases());
         }
+
+        public function testTraitUseIsNotDetectedAsNamespaceUse()
+        {
+            $fixture = new ContextFactory();
+
+            $php = "<?php
+                namespace Foo;
+
+                trait FooTrait {}
+
+                class FooClass {
+                    use FooTrait;
+                }
+            ";
+
+            $fixture = new ContextFactory();
+            $context = $fixture->createForNamespace('Foo', $php);
+
+            $this->assertSame([], $context->getNamespaceAliases());
```

Comment:
You may want to add a test where one trait is imported and used (from a different namespace)